src/components/LayerUI.tsx

```diff
@@ -298,9 +298,31 @@ const LibraryMenu = ({
     },
     [onAddToLibrary],
   );
+  const handleLibraryMenuOnDrop = (event: React.DragEvent<HTMLDivElement>) => {
+    const file = event.dataTransfer.files[0];
+    if (
+      file?.type === MIME_TYPES.excalidrawlib ||
+      file?.name?.endsWith(".excalidrawlib")
+    ) {
+      Library.importLibrary(file)
+        .then(() => {
+          // close and then open to get the libraries updated
+          setAppState({ isLibraryOpen: false });
+          setAppState({ isLibraryOpen: true });
+        })
+        .catch((error) =>
+          setAppState({ isLoading: false, errorMessage: error.message }),
+        );
+    }
+  };
 
   return loadingState === "preloading" ? null : (
-    <Island padding={1} ref={ref} className="layer-ui__library">
+    <Island
+      padding={1}
+      ref={ref}
+      className="layer-ui__library"
+      onDrop={handleLibraryMenuOnDrop}
+    >
```

Comment:
Actually, can we do this in a separate PR to keep the PRs self-contained (not sure why I said that in the issue). Mainly because I just realized we should maybe put the drop handler on the whole excalidraw container instead of having multiple, but it'll be a more major change so best leave it for a separate PR.

Thanks!

Reply:
Sure thing, I'll revert this change. If possible I would like to be assigned to that issue too!